app/response_processor.py

```diff
@@ -98,53 +96,66 @@ def send_receipt(self, decrypted_json):
             self.logger.error("No survey id",
                               tx_id=decrypted_json['tx_id'],
                               ru_ref=decrypted_json['metadata']['ru_ref'])
-            queue_ok = False
+            raise BadMessageError
 
         elif decrypted_json.get("survey_id") == "census":
             self.logger.info("About to publish receipt into ctp queue")
-            queue_ok = self.ctp_publisher.publish_message(dumps(receipt_json), secret=settings.SDX_COLLECT_SECRET)
+
+            try:
+                self.ctp_publisher.publish_message(dumps(receipt_json),
+                                                   secret=settings.SDX_COLLECT_SECRET)
+            except PublishMessageError as e:
+                self.logger.error("Unsuccesful publish", error=e)
+                raise RetryableError
+
         else:
-            self.logger.info("About to publish receipt into rrm queue")
-            queue_ok = self.rrm_publisher.publish_message(dumps(receipt_json), secret=settings.SDX_COLLECT_SECRET)
 
-        if not queue_ok:
-            raise RetryableError()
+            try:
+                self.logger.info("About to publish receipt into rrm queue")
+                self.rrm_publisher.publish_message(dumps(receipt_json),
+                                                   secret=settings.SDX_COLLECT_SECRET)
+            except PublishMessageError as e:
+                self.logger.error("Unsuccesful publish", error=e)
+                raise RetryableError
 
         self.logger.info("Receipt published")
 
     def decrypt_survey(self, encrypted_survey):
         self.logger.info("Decrypting survey")
-        response = self.remote_call(settings.SDX_DECRYPT_URL, data=encrypted_survey)
-        try:
-            self.response_ok(response)
-        except BadMessageError:
-            # Translate the Bad Message into a Decrypt Error to force quarantine
-            raise DecryptError
```

Comment:
Yup still need this - otherwise the next step attempts to read data from the decoded JSON (that'll fail naturally) and cause the message to be rejected altogether rather than quarantined.